keras/src/backend/openvino/trainer.py

```diff
@@ -0,0 +1,272 @@
+            parametrize_data = dict()
+            for elem_name, elem in data.items():
+                param_elem = self._parameterize_data(elem)
+                parametrize_data[elem_name] = param_elem
+        elif isinstance(data, np.ndarray) or np.isscalar(data):
+            ov_type = OPENVINO_DTYPES[str(data.dtype)]
+            ov_shape = list(data.shape)
+            param = ov_opset.parameter(shape=ov_shape, dtype=ov_type)
+            parametrize_data = OpenVINOKerasTensor(param.output(0))
+        elif isinstance(data, int):
+            param = ov_opset.parameter(shape=[], dtype=ov.Type.i32)
+            parametrize_data = OpenVINOKerasTensor(param.output(0))
+        elif isinstance(data, float):
+            param = ov_opset.parameter(shape=[], dtype=ov.Type.f32)
+            parametrize_data = OpenVINOKerasTensor(param.output(0))
+        else:
+            raise "Unknown type of input data {}".format(type(data))
+        return parametrize_data
+
+    def _get_compiled_model(self, data):
+        if (
+            self.ov_compiled_model is not None
+            and get_device() == self.ov_device
+        ):
+            return self.ov_compiled_model
+
+        # remove the previous cached compiled model if exists
+        del self.ov_compiled_model
+
+        # prepare parameterized input
+        self.struct_params = self._parameterize_data(data)
+        # construct OpenVINO graph during calling Keras Model
+        self.struct_outputs = self(self.struct_params)
+
+        parameters = []
+        for p in tree.flatten(self.struct_params):
+            parameters.append(p.output.get_node())
+        results = []
+        for r in tree.flatten(self.struct_outputs):
+            results.append(ov_opset.result(r.output))
+
+        # prepare compiled model from scratch
+        ov_model = ov.Model(results=results, parameters=parameters)
+        self.ov_compiled_model = ov.compile_model(ov_model, get_device())
+        self.ov_device = get_device()
+        return self.ov_compiled_model
+
+    def make_predict_function(self, force=False):
+        if self.predict_function is not None and not force:
+            return self.predict_function
+
+        def one_predict_step(data):
+            data = data[0]
+            return self.predict_step(data)
+
+        def multi_predict_steps(data):
+            outputs = one_predict_step(data[:1])
+
+            for single_step_data in data[1:]:
+                step_outputs = one_predict_step([single_step_data])
+                outputs = tree.map_structure(
+                    lambda t1, t2: np.concatenate([t1, t2]),
+                    outputs,
+                    step_outputs,
+                )
+            return outputs
+
+        if self.steps_per_execution > 1:
+            predict_step = multi_predict_steps
+        else:
+            predict_step = one_predict_step
+
+        self.predict_function = predict_step
+
+    def fit(
+        self,
```

Comment:
I assume all these args can be replaced with `*, **` since they're unused

Reply:
It is done with the same style as for numpy backend to be aligned so may be it is better to leave unchanged.

Reply:
Thanks sounds reasonable